bindgen/ir.ml

```diff
@@ -44,8 +45,17 @@ module Lift = struct
         let rec_fields = List.map lift_record_field record.fields in
         Some (Ir_type (Record { rec_name = record.name; rec_fields }))
 
-  let lift_enum name _constants _complete_definition _attributes =
-    if name = "" then None else Some (Ir_type (Enum { enum_name = name }))
+  let lift_enum name (constants : Clang.Ast.enum_constant list)
+      _complete_definition _attributes =
+    let ir_variants : ir_enum_variant list =
+      List.mapi
+        (fun i (constant : Clang.Ast.enum_constant) ->
+          let desc = constant.desc in
+          { variant_name = desc.constant_name; constant = i })
```

Comment:
Do we know if the C names are always gonna make valid OCaml variant constructors? Otherwise maybe we can fix this in `caml.ml` by processing the name there?

Reply:
So... been reading this: https://www.open-std.org/jtc1/sc22/wg14/www/docs/n1570.pdf

- C can have any digit (0-9), nondigit (_, a-zA-Z) but must start with a nondigit -> This seems to match with OCaml 
- There are also some restrictions in Annex D on some diacritic and sup/subscript unicodes that are not allowed - Not sure how that maps to OCaml; I can investigate further.
- I'll test some different enum naming conventions in the C now.

Reply:
Current behaviour is now
```
C enum prefixed with underscore(s)? -> remove unless 'prefix_underscore' is provided, in which case replace with "Under" (default off)
C enum starting with a lowercase non-digit -> capitalize OCaml variant constructor
```